2-Draft-Accepted/RFCxxxx-PSCredential-Improvements.md

```diff
@@ -0,0 +1,102 @@
+---
+RFC: RFCxxxx
+Author: Steve Lee
+Status: Draft
+SupercededBy: N/A
+Version: 0.1
+Area: Security
+Comments Due: June 30th, 2019
+Plan to implement: Yes, PS7
+---
+
+# Simplify and secure use of credentials in scripts
+
+Advanced scripts that touch many systems require multiple credentials and
+types of credentials particularly when orchestrating across different clouds.
+Best practice is to not hard code any credentials in scripts, but currently
+this requires custom code on different platforms to handle this securely.
+
+This RFC proposes new cmdlets to make this simpler and secure.
+
+## Motivation
+
+> As a PowerShell script author,
+> I can securely use multiple credentials,
+> so that I can automate complex orchestration across multiple remote resources.
+
+## User Experience
+
+Registering credentials for later use:
+
+```powershell
+Install-Module AzVault
+New-CredentialVault -Name Azure -Extension AzVault -Credential $azCred
+$cred = Get-Credential -AccessToken $token
+Register-Credential -Credential $cred -Name AzDevOps -Vault Azure
+$azDevOpsCred = Get-Credential -Name AzDevOps
+Invoke-RestMethod https://dev.azure.com/... -Credential $azDevOpsCred -Authentication Basic
+```
+
+## Specification
+
+### Credential Vaults
+
+There are two types of vaults: local and remote.
+A local vault, by default, is already created and named `Default`.
+`Default` on Windows is [CredMan](https://docs.microsoft.com/en-us/windows/desktop/SecAuthN/credentials-management).
+`Default` on macOS is [KeyChain](https://developer.apple.com/documentation/security/keychain_services).
+
+>[!NOTE]
+>KeyChain support is unlikely to be available in the first release of this feature.
+
+`Default` on Linux is [GNOME Keyring](https://wiki.gnome.org/Projects/GnomeKeyring/).
```

Comment:
We may want to choose one that is tied to a server and not a client distro.

Reply:
I found this answer that has a list and I've heard Hashicorp's vault mentioned at a conference
https://serverfault.com/questions/26804/password-management-system-for-multiple-sysadmins

Reply:
Will update to Hashicorp Vault